lib/src/date-input/DateInput.tsx

```diff
@@ -191,7 +197,7 @@ const DxcDateInput = React.forwardRef<RefType, DateInputPropsType>(
                 align="end"
                 aria-modal={true}
                 onBlur={handleDatePickerOnBlur}
-                onEscapeKeyDown={handleDatePickerEscKeydown}
```

Comment:
This might seem redundant, however it is important because, from my testing, it seems that `onEscapeKeyDown` has a 'lower priority' than `onKeyDown`, making it being triggered *after* the parent's `onKeyDown`, which in our case closes the whole modal.